system/globalid/lnuid.cpp

```diff
@@ -87,7 +89,11 @@ namespace ln_uid {
 #if defined(__linux__) || defined(__APPLE__)
         FILE *fp;
         fp = fopen("/dev/urandom", "r");
-        fread(&randomdata, 1, random_byte_count, fp);
+        if (fread(&randomdata, 1, random_byte_count, fp) != random_byte_count)
```

Comment:
ok, a little paranoid perhaps, but should check fp is not NULL.
And then only fclose() in that case also.

Reply:
Why not use the getrandom() instead ?

Reply:
ok, perhaps older glibc does not have getrandom() call ...